### PR TITLE
[agw][lte] Ipv6 dedicated bearer support

### LINKED ISSUE #3423: IPv6 dedicated bearer support

### PR DESCRIPTION
**Pull request by pruthvihebbani**
[agw][lte] Ipv6 dedicated bearer support

## Summary

This PR contains :
1.	Generation of DL flow rules required for ipv4v6 dedicated tunnel creation 
2.	Modification of ovs rule verification utility to support ipv4v6
3.	New s1 sim test cases to test dedicated bearer for ipv4v6
4.	Incorporation of review comments that were given on PR #3700

## Test Plan
- Verified s1 sim sanity and ipv6 testcases

Note- This PR has dependency on s1ap tester PR #41

### CODE REVIEW

File: lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp [hunk 1]
```diff
@@ -228,7 +228,7 @@ static void add_downlink_match_ipv6(
   /* TODO-Made this local fix as it was not yet available in master
    * without this fix ovs rules are not getting created
    */
-  of13::EthType ip6_type(0x086DD);
+  of13::EthType ip6_type(0x86DD);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp:231]: why is this reverted?

File: lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp [hunk 2]
```diff
@@ -265,7 +265,7 @@ static void add_ded_brr_dl_match(
       downlink_fm.add_oxm_field(ipv4_src);
     }
   } else {
-    of13::EthType ip_type(0x086DD);
+    of13::EthType ip_type(0x86DD);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp:268]: why is this reverted?

**Review by ulaskozat — CHANGES_REQUESTED** [review]
Please include the fix for the dedicated bearer rule set up in GTP Application, e.g., see the following patch:

```
diff --git a/lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp b/lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp
index 8c15f587b..1d965fdb2 100644
--- a/lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp
+++ b/lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp
@@ -269,7 +269,14 @@ static void add_ded_brr_dl_match(
     downlink_fm.add_oxm_field(ip_type);
 
     if (flow.set_params & DST_IPV6) {
-      of13::IPv6Dst ipv6_dst(IPAddress(flow.dst_ip6));
+      // Match on the prefix portion -- this is UE IPv6 address
+      static IPAddress mask("ffff:ffff:ffff:ffff::");
+      struct in6_addr dst_ip6_masked;
+      mask_ipv6_address(
+          (uint8_t*) &dst_ip6_masked, (const uint8_t*) &flow.dst_ip6,
+          mask.getIPv6());
+
+      of13::IPv6Dst ipv6_dst(IPAddress(dst_ip6_masked), mask);
       downlink_fm.add_oxm_field(ipv6_dst);
     }
```
fix the test case accordingly.

File: lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp (hunk 1 above)
Comment (pruthvihebbani) [reply lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp:231]: It should be 0x86DD and not 0x086DD. 0 is redundant

File: lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp (hunk 2 above)
Comment (pruthvihebbani) [reply lte/gateway/c/oai/lib/openflow/controller/GTPApplication.cpp:268]: It should be 0x86DD and not 0x086DD. 0 is redundant

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**